Source/FieldSolver/FiniteDifferenceSolver/ApplySilverMuellerBoundary.cpp

```diff
@@ -33,6 +32,88 @@ void FiniteDifferenceSolver::ApplySilverMuellerBoundary (
     // Ensure that we are using the cells the domain
     domain_box.enclosedCells();
 
+#ifdef WARPX_DIM_RZ
+    // Calculate relevant coefficients
+    amrex::Real const cdt = PhysConst::c*dt;
+    amrex::Real const cdt_over_dr = cdt*m_stencil_coefs_r[0];
+    amrex::Real const coef1_r = (1._rt - cdt_over_dr)/(1._rt + cdt_over_dr);
+    amrex::Real const coef2_r = 2._rt*cdt_over_dr/(1._rt + cdt_over_dr) / PhysConst::c;
+    amrex::Real const coef3_r = cdt/(1._rt + cdt_over_dr) / PhysConst::c;
+
+    // Extract stencil coefficients
+    Real const * const AMREX_RESTRICT coefs_z = m_stencil_coefs_z.dataPtr();
+    int const n_coefs_z = m_stencil_coefs_z.size();
+
+    // Extract cylindrical specific parameters
+    Real const dr = m_dr;
+    int const nmodes = m_nmodes;
+    Real const rmin = m_rmin;
+
+    // tiling is usually set by TilingIfNotGPU()
+    // but here, we set it to false because of potential race condition,
+    // since we grow the tiles by one guard cell after creating them.
+    for ( MFIter mfi(*Efield[0], false); mfi.isValid(); ++mfi ) {
+        // Extract field data for this grid/tile
+        Array4<Real> const& Er = Efield[0]->array(mfi);
+        Array4<Real> const& Et = Efield[1]->array(mfi);
+        Array4<Real> const& Ez = Efield[2]->array(mfi);
+        // Array4<Real> const& Br = Bfield[0]->array(mfi);
+        Array4<Real> const& Bt = Bfield[1]->array(mfi);
+        Array4<Real> const& Bz = Bfield[2]->array(mfi);
+
+        // Extract tileboxes for which to loop
+        Box tbr  = mfi.tilebox(Bfield[0]->ixType().toIntVect());
+        Box tbt  = mfi.tilebox(Bfield[1]->ixType().toIntVect());
+        Box tbz  = mfi.tilebox(Bfield[2]->ixType().toIntVect());
+
+        // We will modify the first (i.e. innermost) guard cell
+        // (if it is outside of the physical domain)
+        // Thus, the tileboxes here are grown by 1 guard cell
+        tbr.grow(1);
+        tbt.grow(1);
+        tbz.grow(1);
+
+        // Loop over the cells
+        amrex::ParallelFor(tbr, tbt, tbz,
+            [=] AMREX_GPU_DEVICE (int /*i*/, int /*j*/, int /*k*/){},
```

Comment:
```suggestion
        amrex::ParallelFor(tbt, tbz,
```

Reply:
Thank you for the 2 box `ParallelFor` suggestion, it's much cleaner and more efficient.
I made this correction and removed `tbr` since we don't use it anymore.